error log when re-init checkpoint for load unit

## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do? If possible, provide a recipe for reproducing the error.

2. What did you expect to see?

3. What did you see instead?

    ```
    [2020/09/08 13:38:21.619 +08:00] [ERROR] [baseconn.go:182] ["execute statement failed"] [task=dm-bi-4304-2] [unit=load] [query="CREATE TABLE `lbs_poi_price` (`poi_id` varchar(64) NOT NULL,`price` double(11,8) NOT NULL,PRIMARY KEY (`poi_id`)) ENGINE=MyISAM DEFAULT CHARSET=utf8 COMMENT='POI-PRICE';"] [argument="[]"] [error="Error 1050: Table 'lbs.lbs_poi_price' already exists"]
    [2020/09/08 13:38:21.620 +08:00] [ERROR] [db.go:176] ["execute statements failed after retry"] [task=dm-bi-4304-2] [unit=load] [queries="[USE `lbs`; CREATE TABLE `lbs_poi_price` (`poi_id` varchar(64) NOT NULL,`price` double(11,8) NOT NULL,PRIMARY KEY (`poi_id`)) ENGINE=MyISAM DEFAULT CHARSET=utf8 COMMENT='POI-PRICE';]"] [arguments="[]"] [error="[code=10006:class=database:scope=not-set:level=high], Message: execute statement failed: CREATE TABLE `lbs_poi_price` (`poi_id` varchar(64) NOT NULL,`price` double(11,8) NOT NULL,PRIMARY KEY (`poi_id`)) ENGINE=MyISAM DEFAULT CHARSET=utf8 COMMENT='POI-PRICE';, RawCause: Error 1050: Table 'lbs.lbs_poi_price' already exists"]
    [2020/09/08 13:38:21.620 +08:00] [INFO] [loader.go:1019] ["table already exists, skip it"] [task=dm-bi-4304-2] [unit=load] ["table schema file"=./dumped_data.dm-bi-4304-2/lbs.lbs_poi_price-schema.sql]
    ```

    ```
    [2020/09/08 12:33:27.717 +08:00] [ERROR] [baseconn.go:182] ["execute statement failed"] [task=dm-bi-4304-2] [unit=load] ["worker ID"=11] [query="INSERT INTO `dm_meta`.`dm-bi-4304-2_loader_checkpoint` (`id`, `filename`, `cp_schema`, `cp_table`, `offset`, `end_pos`) VALUES(?,?,?,?,?,?)"] [argument="[4304-replica-02 lbs.lbs_monitor_action.5.sql lbs lbs_monitor_action 0 97364896]"] [error="Error 1062: Duplicate entry '4304-replica-02-lbs.lbs_monitor_action.5.sql' for key 'uk_id_f'"]
    [2020/09/08 12:33:27.717 +08:00] [ERROR] [db.go:176] ["execute statements failed after retry"] [task=dm-bi-4304-2] [unit=load] ["worker ID"=11] [queries="[INSERT INTO `dm_meta`.`dm-bi-4304-2_loader_checkpoint` (`id`, `filename`, `cp_schema`, `cp_table`, `offset`, `end_pos`) VALUES(?,?,?,?,?,?)]"] [arguments="[[4304-replica-02 lbs.lbs_monitor_action.5.sql lbs lbs_monitor_action 0 97364896]]"] [error="[code=10006:class=database:scope=not-set:level=high], Message: execute statement failed: INSERT INTO `dm_meta`.`dm-bi-4304-2_loader_checkpoint` (`id`, `filename`, `cp_schema`, `cp_table`, `offset`, `end_pos`) VALUES(?,?,?,?,?,?), RawCause: Error 1062: Duplicate entry '4304-replica-02-lbs.lbs_monitor_action.5.sql' for key 'uk_id_f'"]
    [2020/09/08 12:33:27.717 +08:00] [INFO] [checkpoint.go:307] ["checkpoint record already exists, skip it."] [task=dm-bi-4304-2] [unit=load] [component="remote checkpoint"] [id=4304-replica-02] [filename=lbs.lbs_monitor_action.5.sql]
    ```

    if checkpoint for the table `already exists`, we can `skip it`, but we log some `ERROR` and it's misleading.

4. Versions of the cluster

    - DM version (run `dmctl -V` or `dm-worker -V` or `dm-master -V`):

        ```
        v2.0.0-rc
        ```

--- a/loader/checkpoint.go
+++ b/loader/checkpoint.go
@@ -312,18 +312,12 @@ func (cp *RemoteCheckPoint) AllFinished() bool {
 
 // Init implements CheckPoint.Init
 func (cp *RemoteCheckPoint) Init(tctx *tcontext.Context, filename string, endPos int64) error {
-	idx := strings.Index(filename, ".sql")
-	if idx < 0 {
-		return terror.ErrCheckpointInvalidTableFile.Generate(filename)
-	}
-	fname := filename[:idx]
-	fields := strings.Split(fname, ".")
-	if len(fields) != 2 && len(fields) != 3 {
-		return terror.ErrCheckpointInvalidTableFile.Generate(filename)
-	}
-
 	// fields[0] -> db name, fields[1] -> table name
-	schema, table := fields[0], fields[1]
+	schema, table, err := getDBAndTableFromFilename(filename)
+	if err != nil {
+		return terror.ErrCheckpointInvalidTableFile.Generate(filename)
+
+	}
 	sql2 := fmt.Sprintf("INSERT INTO %s (`id`, `filename`, `cp_schema`, `cp_table`, `offset`, `end_pos`) VALUES(?,?,?,?,?,?)", cp.tableName)
 	cp.logCtx.L().Info("initial checkpoint record",
 		zap.String("sql", sql2),
@@ -335,7 +329,7 @@ func (cp *RemoteCheckPoint) Init(tctx *tcontext.Context, filename string, endPos
 		zap.Int64("end position", endPos))
 	args := []interface{}{cp.id, filename, schema, table, 0, endPos}
 	cp.connMutex.Lock()
-	err := cp.conn.executeSQL(tctx, []string{sql2}, args)
+	err = cp.conn.executeSQL(tctx, []string{sql2}, args)
 	cp.connMutex.Unlock()
 	if err != nil {
 		if isErrDupEntry(err) {
@@ -383,11 +377,11 @@ func (cp *RemoteCheckPoint) GenSQL(filename string, offset int64) string {
 
 // UpdateOffset implements CheckPoint.UpdateOffset
 func (cp *RemoteCheckPoint) UpdateOffset(filename string, offset int64) {
-	fields := strings.Split(filename, ".")
-	if len(fields) != 2 && len(fields) != 3 {
-		cp.logCtx.L().Error("can't get db and table from filename in checkpoint UpdateOffset", zap.String("filename", filename))
+	db, table, err := getDBAndTableFromFilename(filename)
+	if err != nil {
+		cp.logCtx.L().Error("error in checkpoint UpdateOffset", zap.Error(err))
+		return
 	}
-	db, table := fields[0], fields[1]
 	cp.restoringFiles[db][table][filename][0] = offset
 }
 

--- a/loader/checkpoint.go
+++ b/loader/checkpoint.go
@@ -44,6 +44,9 @@ type CheckPoint interface {
 	// GetAllRestoringFileInfo return all restoring files position
 	GetAllRestoringFileInfo() map[string][]int64
 
+	// IsTableCreated checks if db / table was created. set `table` to "" when check db
+	IsTableCreated(db, table string) bool
+
 	// IsTableFinished query if table has finished
 	IsTableFinished(db, table string) bool
 
@@ -67,6 +70,13 @@ type CheckPoint interface {
 
 	// GenSQL generates sql to update checkpoint to DB
 	GenSQL(filename string, offset int64) string
+
+	// UpdateOffset keeps `cp.restoringFiles` in memory same with checkpoint in DB,
+	// should be called after update checkpoint in DB
+	UpdateOffset(filename string, offset int64)
+
+	// AllFinished returns `true` when all restoring job are finished
+	AllFinished() bool
 }
 
 // RemoteCheckPoint implements CheckPoint by saving status in remote database system, mostly in TiDB.
@@ -80,8 +90,8 @@ type RemoteCheckPoint struct {
 	conn           *DBConn
 	id             string
 	schema         string
-	tableName      string // tableName contains schema name
-	restoringFiles map[string]map[string]FilePosSet
+	tableName      string                           // tableName contains schema name
+	restoringFiles map[string]map[string]FilePosSet // schema -> table -> FilePosSet(filename -> [cur, end])
 	finishedTables map[string]struct{}
 	logCtx         *tcontext.Context
 }
@@ -219,6 +229,19 @@ func (cp *RemoteCheckPoint) GetAllRestoringFileInfo() map[string][]int64 {
 	return results
 }
 
+// IsTableCreated implements CheckPoint.IsTableCreated
+func (cp *RemoteCheckPoint) IsTableCreated(db, table string) bool {
+	tables, ok := cp.restoringFiles[db]
+	if !ok {
+		return false
+	}
+	if table == "" {
+		return true
+	}
+	_, ok = tables[table]
+	return ok
+}
+
 // IsTableFinished implements CheckPoint.IsTableFinished
 func (cp *RemoteCheckPoint) IsTableFinished(db, table string) bool {
 	key := strings.Join([]string{db, table}, ".")
@@ -258,7 +281,7 @@ func (cp *RemoteCheckPoint) CalcProgress(allFiles map[string]Tables2DataFiles) e
 		}
 	}
 
-	cp.logCtx.L().Info("calculate checkpoint finished.", zap.Reflect("finished tables", cp.finishedTables))
+	cp.logCtx.L().Info("calculate checkpoint finished.", zap.Any("finished tables", cp.finishedTables))
 	return nil
 }
 
@@ -270,6 +293,18 @@ func (cp *RemoteCheckPoint) allFilesFinished(files map[string][]int64) bool {
 		}
 		if pos[0] != pos[1] {
 			return false
+		}
+	}
+	return true
+}
+
+// AllFinished implements CheckPoint.AllFinished
+func (cp *RemoteCheckPoint) AllFinished() bool {
+	for _, tables := range cp.restoringFiles {
+		for _, restoringFiles := range tables {
+			if !cp.allFilesFinished(restoringFiles) {
+				return false
+			}
 		}
 	}
 	return true
@@ -304,7 +339,7 @@ func (cp *RemoteCheckPoint) Init(tctx *tcontext.Context, filename string, endPos
 	cp.connMutex.Unlock()
 	if err != nil {
 		if isErrDupEntry(err) {
-			cp.logCtx.L().Info("checkpoint record already exists, skip it.", zap.String("id", cp.id), zap.String("filename", filename))
+			cp.logCtx.L().Error("checkpoint record already exists, skip it.", zap.String("id", cp.id), zap.String("filename", filename))
 			return nil
 		}
 		return terror.WithScope(terror.Annotate(err, "initialize checkpoint"), terror.ScopeDownstream)
@@ -344,6 +379,16 @@ func (cp *RemoteCheckPoint) GenSQL(filename string, offset int64) string {
 	sql := fmt.Sprintf("UPDATE %s SET `offset`=%d WHERE `id` ='%s' AND `filename`='%s';",
 		cp.tableName, offset, cp.id, filename)
 	return sql
+}
+
+// UpdateOffset implements CheckPoint.UpdateOffset
+func (cp *RemoteCheckPoint) UpdateOffset(filename string, offset int64) {
+	fields := strings.Split(filename, ".")
+	if len(fields) != 2 && len(fields) != 3 {
+		cp.logCtx.L().Error("can't get db and table from filename in checkpoint UpdateOffset", zap.String("filename", filename))
+	}
+	db, table := fields[0], fields[1]
+	cp.restoringFiles[db][table][filename][0] = offset
 }
 
 // Clear implements CheckPoint.Clear

--- a/loader/checkpoint_test.go
+++ b/loader/checkpoint_test.go
@@ -70,6 +70,14 @@ func (t *testCheckPointSuite) TestForDB(c *C) {
 		{"db1.tbl3.sql", 789},
 	}
 
+	allFiles := map[string]Tables2DataFiles{
+		"db1": {
+			"tbl1": {cases[0].filename},
+			"tbl2": {cases[1].filename},
+			"tbl3": {cases[2].filename},
+		},
+	}
+
 	id := "test_for_db"
 	tctx := tcontext.Background()
 	cp, err := newRemoteCheckPoint(tctx, t.cfg, id)
@@ -89,11 +97,25 @@ func (t *testCheckPointSuite) TestForDB(c *C) {
 	c.Assert(err, IsNil)
 	c.Assert(count, Equals, 0)
 
+	c.Assert(cp.IsTableCreated("db1", ""), IsFalse)
+	c.Assert(cp.IsTableCreated("db1", "tbl1"), IsFalse)
+	c.Assert(cp.CalcProgress(allFiles), IsNil)
+	c.Assert(cp.IsTableFinished("db1", "tbl1"), IsFalse)
+
 	// insert default checkpoints
 	for _, cs := range cases {
 		err = cp.Init(tctx, cs.filename, cs.endPos)
 		c.Assert(err, IsNil)
 	}
+
+	c.Assert(cp.IsTableCreated("db1", ""), IsTrue)
+	c.Assert(cp.IsTableCreated("db1", "tbl1"), IsTrue)
+	c.Assert(cp.CalcProgress(allFiles), IsNil)
+	c.Assert(cp.IsTableFinished("db1", "tbl1"), IsFalse)
+
+	info := cp.GetRestoringFileInfo("db1", "tbl1")
+	c.Assert(info, HasLen, 1)
+	c.Assert(info[cases[0].filename], DeepEquals, []int64{0, cases[0].endPos})
 
 	err = cp.Load(tctx)
 	c.Assert(err, IsNil)
@@ -129,6 +151,15 @@ func (t *testCheckPointSuite) TestForDB(c *C) {
 	err = cp.Load(tctx)
 	c.Assert(err, IsNil)
 
+	c.Assert(cp.IsTableCreated("db1", ""), IsTrue)
+	c.Assert(cp.IsTableCreated("db1", "tbl1"), IsTrue)
+	c.Assert(cp.CalcProgress(allFiles), IsNil)
+	c.Assert(cp.IsTableFinished("db1", "tbl1"), IsTrue)
+
+	info = cp.GetRestoringFileInfo("db1", "tbl1")
+	c.Assert(info, HasLen, 1)
+	c.Assert(info[cases[0].filename], DeepEquals, []int64{cases[0].endPos, cases[0].endPos})
+
 	infos = cp.GetAllRestoringFileInfo()
 	c.Assert(len(infos), Equals, len(cases))
 	for _, cs := range cases {
@@ -149,6 +180,11 @@ func (t *testCheckPointSuite) TestForDB(c *C) {
 	// no checkpoint exist
 	err = cp.Load(tctx)
 	c.Assert(err, IsNil)
+
+	c.Assert(cp.IsTableCreated("db1", ""), IsFalse)
+	c.Assert(cp.IsTableCreated("db1", "tbl1"), IsFalse)
+	c.Assert(cp.CalcProgress(allFiles), IsNil)
+	c.Assert(cp.IsTableFinished("db1", "tbl1"), IsFalse)
 
 	infos = cp.GetAllRestoringFileInfo()
 	c.Assert(len(infos), Equals, 0)

--- a/loader/loader.go
+++ b/loader/loader.go
@@ -921,15 +921,11 @@ func (l *Loader) prepareDataFiles(files map[string]struct{}) error {
 			continue
 		}
 
-		idx := strings.Index(file, ".sql")
-		name := file[:idx]
-		fields := strings.Split(name, ".")
-		if len(fields) != 2 && len(fields) != 3 {
-			l.logCtx.L().Warn("invalid db table sql file", zap.String("file", file))
+		db, table, err := getDBAndTableFromFilename(file)
+		if err != nil {
+			l.logCtx.L().Warn("invalid db table sql file", zap.String("file", file), zap.Error(err))
 			continue
 		}
-
-		db, table := fields[0], fields[1]
 		if l.skipSchemaAndTable(&filter.Table{Schema: db, Name: table}) {
 			l.logCtx.L().Warn("ignore data file", zap.String("data file", file))
 			continue
@@ -1286,7 +1282,7 @@ func (l *Loader) cleanDumpFiles() {
 		var lastErr error
 		for f := range files {
 			if strings.HasSuffix(f, ".sql") {
-				lastErr = os.Remove(f)
+				lastErr = os.Remove(filepath.Join(l.cfg.Dir, f))
 			}
 		}
 		if lastErr != nil {

--- a/loader/loader.go
+++ b/loader/loader.go
@@ -49,7 +49,8 @@ import (
 )
 
 const (
-	jobCount = 1000
+	jobCount            = 1000
+	uninitializedOffset = -1
 )
 
 // FilePosSet represents a set in mathematics.
@@ -185,21 +186,8 @@ func (w *Worker) run(ctx context.Context, fileJobQueue chan *fileJob, runFatalCh
 					}
 					return
 				}
+				w.loader.checkPoint.UpdateOffset(job.file, job.offset)
 				w.loader.finishedDataSize.Add(job.offset - job.lastOffset)
-
-				if w.cfg.CleanDumpFile {
-					fileInfos := w.checkPoint.GetRestoringFileInfo(job.schema, job.table)
-					if pos, ok := fileInfos[job.file]; ok {
-						if job.offset == pos[1] {
-							w.tctx.L().Info("try to remove loaded dump file", zap.String("data file", job.file))
-							if err := os.Remove(job.absPath); err != nil {
-								w.tctx.L().Warn("error when remove loaded dump file", zap.String("data file", job.file), zap.Error(err))
-							}
-						}
-					} else {
-						w.tctx.L().Warn("file not recorded in checkpoint", zap.String("data file", job.file))
-					}
-				}
 			}
 		}
 	}
@@ -262,22 +250,28 @@ func (w *Worker) dispatchSQL(ctx context.Context, file string, offset int64, tab
 		cur int64
 	)
 
+	baseFile := filepath.Base(file)
+
 	f, err = os.Open(file)
 	if err != nil {
 		return terror.ErrLoadUnitDispatchSQLFromFile.Delegate(err)
 	}
 	defer f.Close()
 
-	finfo, err := f.Stat()
-	if err != nil {
-		return terror.ErrLoadUnitDispatchSQLFromFile.Delegate(err)
-	}
+	// file was not found in checkpoint
+	if offset == uninitializedOffset {
+		offset = 0
 
-	baseFile := filepath.Base(file)
-	err = w.checkPoint.Init(w.tctx.WithContext(ctx), baseFile, finfo.Size())
-	if err != nil {
-		w.tctx.L().Error("fail to initial checkpoint", zap.String("data file", file), zap.Int64("offset", offset), log.ShortError(err))
-		return err
+		finfo, err2 := f.Stat()
+		if err2 != nil {
+			return terror.ErrLoadUnitDispatchSQLFromFile.Delegate(err2)
+		}
+
+		err2 = w.checkPoint.Init(w.tctx.WithContext(ctx), baseFile, finfo.Size())
+		if err2 != nil {
+			w.tctx.L().Error("fail to initial checkpoint", zap.String("data file", file), zap.Int64("offset", offset), log.ShortError(err2))
+			return err2
+		}
 	}
 
 	cur, err = f.Seek(offset, io.SeekStart)
@@ -647,25 +641,6 @@ func (l *Loader) Restore(ctx context.Context) error {
 
 	if err == nil {
 		l.logCtx.L().Info("all data files have been finished", zap.Duration("cost time", time.Since(begin)))
-		if l.cfg.CleanDumpFile {
-			files := CollectDirFiles(l.cfg.Dir)
-			hasDatafile := false
-			for file := range files {
-				if strings.HasSuffix(file, ".sql") &&
-					!strings.HasSuffix(file, "-schema.sql") &&
-					!strings.HasSuffix(file, "-schema-create.sql") &&
-					!strings.Contains(file, "-schema-view.sql") &&
-					!strings.Contains(file, "-schema-triggers.sql") &&
-					!strings.Contains(file, "-schema-post.sql") {
-					hasDatafile = true
-				}
-			}
-
-			if !hasDatafile {
-				l.logCtx.L().Info("clean dump files after importing all files")
-				l.cleanDumpFiles()
-			}
-		}
 	} else if errors.Cause(err) != context.Canceled {
 		return err
 	}
@@ -693,6 +668,9 @@ func (l *Loader) Close() {
 	err := l.toDB.Close()
 	if err != nil {
 		l.logCtx.L().Error("close downstream DB error", log.ShortError(err))
+	}
+	if l.cfg.CleanDumpFile && l.checkPoint.AllFinished() {
+		l.cleanDumpFiles()
 	}
 	l.checkPoint.Close()
 	l.removeLabelValuesWithTaskInMetrics(l.cfg.Name)
@@ -1034,6 +1012,10 @@ func (l *Loader) prepare() error {
 
 // restoreSchema creates schema
 func (l *Loader) restoreSchema(ctx context.Context, conn *DBConn, sqlFile, schema string) error {
+	if l.checkPoint.IsTableCreated(schema, "") {
+		l.logCtx.L().Info("database already exists in checkpoint, skip creating it", zap.String("schema", schema), zap.String("db schema file", sqlFile))
+		return nil
+	}
 	err := l.restoreStructure(ctx, conn, sqlFile, schema, "")
 	if err != nil {
 		if isErrDBExists(err) {
@@ -1047,6 +1029,10 @@ func (l *Loader) restoreSchema(ctx context.Context, conn *DBConn, sqlFile, schem
 
 // restoreTable creates table
 func (l *Loader) restoreTable(ctx context.Context, conn *DBConn, sqlFile, schema, table string) error {
+	if l.checkPoint.IsTableCreated(schema, table) {
+		l.logCtx.L().Info("table already exists in checkpoint, skip creating it", zap.String("schema", schema), zap.String("table", table), zap.String("db schema file", sqlFile))
+		return nil
+	}
 	err := l.restoreStructure(ctx, conn, sqlFile, schema, table)
 	if err != nil {
 		if isErrTableExists(err) {
@@ -1107,7 +1093,6 @@ func (l *Loader) restoreStructure(ctx context.Context, conn *DBConn, sqlFile str
 				return terror.WithScope(err, terror.ScopeDownstream)
 			}
 		}
-
 	}
 
 	return nil
@@ -1225,7 +1210,7 @@ func (l *Loader) restoreData(ctx context.Context) error {
 
 				l.logCtx.L().Debug("dispatch data file", zap.String("schema", db), zap.String("table", table), zap.String("data file", file))
 
-				var offset int64
+				offset := int64(uninitializedOffset)
 				posSet, ok := restoringFiles[file]
 				if ok {
 					offset = posSet[0]
@@ -1289,24 +1274,23 @@ func (l *Loader) getMydumpMetadata() error {
 
 // cleanDumpFiles is called when finish restoring data, to clean useless files
 func (l *Loader) cleanDumpFiles() {
+	l.logCtx.L().Info("clean dump files")
 	if l.cfg.Mode == config.ModeFull {
 		// in full-mode all files won't be need in the future
 		if err := os.RemoveAll(l.cfg.Dir); err != nil {
 			l.logCtx.L().Warn("error when remove loaded dump folder", zap.String("data folder", l.cfg.Dir), zap.Error(err))
 		}
 	} else {
-		// leave metadata file, only delete structure files
-		for db, tables := range l.db2Tables {
-			dbFile := fmt.Sprintf("%s/%s-schema-create.sql", l.cfg.Dir, db)
-			if err := os.Remove(dbFile); err != nil {
-				l.logCtx.L().Warn("error when remove loaded dump file", zap.String("data file", dbFile), zap.Error(err))
+		// leave metadata file, only delete sql files
+		files := CollectDirFiles(l.cfg.Dir)
+		var lastErr error
+		for f := range files {
+			if strings.HasSuffix(f, ".sql") {
+				lastErr = os.Remove(f)
 			}
-			for table := range tables {
-				tableFile := fmt.Sprintf("%s/%s.%s-schema.sql", l.cfg.Dir, db, table)
-				if err := os.Remove(tableFile); err != nil {
-					l.logCtx.L().Warn("error when remove loaded dump file", zap.String("data file", tableFile), zap.Error(err))
-				}
-			}
+		}
+		if lastErr != nil {
+			l.logCtx.L().Warn("show last error when remove loaded dump sql files", zap.String("data folder", l.cfg.Dir), zap.Error(lastErr))
 		}
 	}
 }

--- a/loader/util.go
+++ b/loader/util.go
@@ -91,3 +91,16 @@ func generateSchemaCreateFile(dir string, schema string) error {
 func escapeName(name string) string {
 	return strings.Replace(name, "`", "``", -1)
 }
+
+// input filename is like `all_mode.t1.0.sql` or `all_mode.t1.sql`
+func getDBAndTableFromFilename(filename string) (string, string, error) {
+	idx := strings.Index(filename, ".sql")
+	if idx < 0 {
+		return "", "", fmt.Errorf("%s doesn't have a `.sql` suffix", filename)
+	}
+	fields := strings.Split(filename[:idx], ".")
+	if len(fields) != 2 && len(fields) != 3 {
+		return "", "", fmt.Errorf("%s doesn't have correct `.` seperator", filename)
+	}
+	return fields[0], fields[1], nil
+}

--- a/tests/all_mode/run.sh
+++ b/tests/all_mode/run.sh
@@ -148,8 +148,8 @@ function run() {
     check_contains "all_mode"
 
     echo "check dump files have been cleaned"
-    ls $WORK_DIR/worker1/dumped_data.test && exit 1 || echo "worker1 auto removed dump files"
-    ls $WORK_DIR/worker2/dumped_data.test && exit 1 || echo "worker2 auto removed dump files"
+    ls $WORK_DIR/worker1/dumped_data.$ILLEGAL_CHAR_NAME && exit 1 || echo "worker1 auto removed dump files"
+    ls $WORK_DIR/worker2/dumped_data.$ILLEGAL_CHAR_NAME && exit 1 || echo "worker2 auto removed dump files"
 
     echo "check no password in log"
     check_log_not_contains $WORK_DIR/master/log/dm-master.log "/Q7B9DizNLLTTfiZHv9WoEAKamfpIUs="

--- a/tests/import_goroutine_leak/run.sh
+++ b/tests/import_goroutine_leak/run.sh
@@ -26,7 +26,6 @@ function run() {
     run_sql_file $WORK_DIR/db1.prepare.sql $MYSQL_HOST1 $MYSQL_PORT1 $MYSQL_PASSWORD1
     run_sql_file $WORK_DIR/db2.prepare.sql $MYSQL_HOST2 $MYSQL_PORT2 $MYSQL_PASSWORD2
 
-
     echo "dm-worker panic, doJob of import unit workers don't exit"
     # check doJobs of import unit worker exit
     inject_points=("github.com/pingcap/dm/loader/dispatchError=return(1)"
@@ -70,6 +69,7 @@ function run() {
     inject_points=("github.com/pingcap/dm/loader/dontWaitWorkerExit=return(1)"
                    "github.com/pingcap/dm/loader/LoadDataSlowDown=sleep(1000)"
                    "github.com/pingcap/dm/loader/executeSQLError=return(1)"
+                   "github.com/pingcap/dm/loader/dispatchError=return(1)"
                    )
     export GO_FAILPOINTS="$(join_string \; ${inject_points[@]})"
     run_dm_worker $WORK_DIR/worker1 $WORKER1_PORT $cur/conf/dm-worker1.toml

